rust/src/dataset.rs

```diff
@@ -944,6 +945,27 @@ impl Dataset {
         }
     }
 
+    pub async fn statistics(&self, index_name: Option<String>) -> Result<String> {
```

Comment:
TBH, the existing index API is a bit messy. I partly wish we could keep the stats method on the index itself, and allow something like: 

```rust
let stats = dataset.load_index("my_index").await?.statistics()?; 
```

But right now we just expose the metadata and not the index itself, so we have to expose these stats as part of `Dataset`.

For simplicity, could we make `index_name` required? Then we don't have to do the JSOn list construction or handle the optional parameter. Users can always request each index in a loop if they want it all.

Reply:
Made `index_name` required.